### PR TITLE
hydrapaper: init at 3.3.2

### PR DESCRIPTION
**Pull request by lachrymaLF**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Sigmanificient** [comment]
Result of `nixpkgs-review pr 300104` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hydrapaper</li>
  </ul>
</details>

**Comment by Sigmanificient** [comment]
When testing the `hydrapaper`:
```
Traceback (most recent call last):
  File "/nix/store/ackbdrbibx46gc0jhy6q2d002b6kkg59-hydrapaper-3.3.2/bin/hydrapaper", line 60, in <module>
    gi.require_version('Gtk', '4.0')
  File "/nix/store/9m7kp3cmkqdd848xz27w0pnc2n3vbmzg-python3-3.11.8-env/lib/python3.11/site-packages/gi/__init__.py", line 126, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace Gtk not available
```

**Comment by lachrymaLF** [comment]
Right, I think it's supposed to be gtk4 instead..

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    glib
+    pkg-config
+    dbus
+    appstream
+    gtk3
+  ];
+
+  buildInputs = [
+    glib
+    gtk3
```
Comment (Lilacious) [inline pkgs/by-name/hy/hydrapaper/package.nix:44]:
```suggestion

```
Please remove, gtk is only required in nativeBuildInputs

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    blueprint-compiler
+    gobject-introspection
+    meson
+    pandoc
+    ninja
+    glib
+    pkg-config
+    dbus
+    appstream
+    gtk3
```
Comment (Lilacious) [inline pkgs/by-name/hy/hydrapaper/package.nix:39]:
```suggestion
    gtk4
    wrapGAppsHook
```
The dependency listed in the source repo is `gtk4`.
Include wrapGAppsHook to fix `ValueError: Namespace Gtk not available`.

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  python3,
+  blueprint-compiler,
+  gobject-introspection,
+  pandoc,
+  libadwaita,
+  glib,
+  pkg-config,
+  dbus,
+  appstream,
+  gtk3
```
Comment (Lilacious) [inline pkgs/by-name/hy/hydrapaper/package.nix:15]:
```suggestion
  gtk4
  wrapGAppsHook
```

**Review by Lilacious — CHANGES_REQUESTED** [review]

**Comment by lachrymaLF** [comment]
Thanks!

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  python3,
+  blueprint-compiler,
+  gobject-introspection,
+  pandoc,
+  libadwaita,
+  glib,
+  pkg-config,
+  dbus,
+  appstream,
+  gtk4
```
Comment (Sigmanificient) [inline pkgs/by-name/hy/hydrapaper/package.nix:15]: missing comma?
```suggestion
  gtk4,
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/hy/hydrapaper/package.nix (hunk 4 above)
Comment (Lilacious) [reply pkgs/by-name/hy/hydrapaper/package.nix:15]: Ah yes, indeed.

**Review by Lilacious — APPROVED** [review]

**Comment by Sigmanificient** [comment]
Result of `nixpkgs-review pr 300104` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hydrapaper</li>
  </ul>
</details>

**Review by Sigmanificient — APPROVED** [review]

**Comment by Sigmanificient** [comment]
`sh: line 1: xdg-user-dirs-update: command not found`

**Comment by Lilacious** [comment]
The gitlab description says:
> A Gtk utility to set different backgrounds for each monitor on GNOME
So I assume that it will not work in any other DEs or WMs.

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -10829,6 +10829,11 @@
     }];
     name = "Yaroslav Bolyukin";
   };
+  lachrymal = {
```
Comment (Aleksanaa) [inline maintainers/maintainer-list.nix:10832]: It's better to add an email although there is no standard yet. It is possible that when you abandon this account, others will still want to communicate with you about changes or removals of maintainer information.

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    blueprint-compiler
+    gobject-introspection
+    meson
+    pandoc
+    ninja
+    glib
+    pkg-config
+    dbus
+    appstream
+    gtk4
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hydrapaper/package.nix:40]:
```suggestion
```

gtk4 is propagated by wrapGApps4Hook

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    wrapGAppsHook
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hydrapaper/package.nix:41]:
```suggestion
    wrapGApps4Hook
```

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+  nativeBuildInputs = [
+    blueprint-compiler
+    gobject-introspection
+    meson
+    pandoc
+    ninja
+    glib
+    pkg-config
+    dbus
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hydrapaper/package.nix:38]: Looks like dbus should be inside `buildInputs`.

File: pkgs/by-name/hy/hydrapaper/package.nix (hunk 7 above)
Comment (Lilacious) [reply pkgs/by-name/hy/hydrapaper/package.nix:41]: I think it is wrapGAppsHook4

File: pkgs/by-name/hy/hydrapaper/package.nix (hunk 8 above)
Comment (lachrymaLF) [reply pkgs/by-name/hy/hydrapaper/package.nix]: Indeed, thank you.

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  pandoc,
+  libadwaita,
+  glib,
+  pkg-config,
+  dbus,
+  appstream,
+  wrapGAppsHook4
+}:
+
+stdenv.mkDerivation {
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hydrapaper/package.nix:18]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  pname = "hydrapaper";
+  version = "3.3.2";
+
+  src = fetchFromGitLab {
+    owner = "gabmus";
+    repo = "HydraPaper";
+    rev = "4589088da390362906ac9fecb8d4b26f594656d9";
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hydrapaper/package.nix:25]:
```suggestion
    rev = finalAttrs.version;
```

Comment (lachrymaLF) [reply pkgs/by-name/hy/hydrapaper/package.nix]: I didn't know about that, thanks!

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    dbus
+    glib
+    libadwaita
+    (python3.withPackages (ps: with ps; [
+      dbus-python
+      pygobject3
+      pillow
+    ]))
+  ];
+
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hydrapaper/package.nix:57]:
```suggestion
preFixup = ''
  gappsWrapperArgs+=(
    --prefix PATH : ${lib.makeBinPath [ glib xdg-user-dirs ]}
  )
'';
```

Didn't wrap `swaymsg` as it is part of `sway` package. We'd assume it is there.

**Review by Aleksanaa — COMMENTED** [review]
Others lgtm. Tested.

**Review by Aleksanaa — COMMENTED** [review]
Wait, no, this should be packaged by `python3Packages.buildPythonApplication`

**Comment by lachrymaLF** [comment]
> Wait, no, this should be packaged by `python3Packages.buildPythonApplication`

It reads to me like `python3Packages.buildPythonApplication` is to be used with `setup.py` based apps. The build system is meson in this case. Does it really make sense to use `buildPythonApplication`?

**Comment by Aleksanaa** [comment]
> It reads to me like `python3Packages.buildPythonApplication` is to be used with `setup.py` based apps. The build system is meson in this case. Does it really make sense to use `buildPythonApplication`?

That is not limited to `pyproject`. See these packages by me:

https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/va/varia/package.nix
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/as/ascii-draw/package.nix#L55

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 12]
```diff
@@ -0,0 +1,78 @@
+{ lib,
+  stdenv,
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hydrapaper/package.nix:2]:
```suggestion
```

File: pkgs/by-name/hy/hydrapaper/package.nix [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+  fetchFromGitLab,
+  meson,
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hydrapaper/package.nix:4]: I would recommend that you sort them in the order they are used, but this is not necessary

**Comment by Aleksanaa** [comment]
Remember to squash your commits next time. This PR is taking too long, so I’ll help you first.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**